File: FinanceApp.Shared/Dto/CurrentBalance/CreateOrUpdateCurrentBalance.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Dto.CurrentBalance
{
    public class CreateOrUpdateCurrentBalance : CreateOrUpdateDto
    {
        public double Value { get; set; }
        public double? PercentageCdi { get; set; }
        public bool UpdateValueWithCdiIndex { get; set; }

    }
}

File: FlutterApp/finance_app/lib/models/current_balance/create_or_update_current_balance.dart
class CreateOrUpdateCurrentBalance {
  // String id;
  double? percentageCdi;
  bool updateValueWithCdiIndex;
  double value;

  CreateOrUpdateCurrentBalance({
    // required this.id,
    required this.percentageCdi,
    required this.value,
    required this.updateValueWithCdiIndex,
  });

  Map<String, dynamic> toJson() => {
        // 'id': id,
        'percentageCdi': percentageCdi,
        'value': value,
        'updateValueWithCdiIndex': updateValueWithCdiIndex.toString(),
      };
}
